Tests/IGListAdapterTests.m

```diff
@@ -1067,6 +1067,24 @@ - (void)test_whenSupplementarySourceReturnsNegativeSize_thatAdapterReturnsZero {
     XCTAssertEqual(size.height, 0.0);
 }
 
+- (void)test_whenSectionIndexGreaterThanZero_thatAdapterReturnsSizeForSectionIndexesGreaterThanZero {
+  self.dataSource.objects = @[@1];
+  [self.adapter reloadDataWithCompletion:nil];
+  
+  IGTestSupplementarySource *supplementarySource = [IGTestSupplementarySource new];
+  supplementarySource.collectionContext = self.adapter;
+  supplementarySource.supportedElementKinds = @[UICollectionElementKindSectionFooter];
+  
+  IGListSectionController *controller = [self.adapter sectionControllerForObject:@1];
+  controller.supplementaryViewSource = supplementarySource;
+  supplementarySource.sectionController = controller;
+  
+  const CGSize size = [self.adapter sizeForSupplementaryViewOfKind:UICollectionElementKindSectionHeader
+                                                       atIndexPath:[NSIndexPath indexPathForItem:0 inSection:1]];
```

Comment:
The code you've linked as an example is exactly the one I used as base for mine lol and I don't see any difference other than my section controller not being the implementer of IGListSupplementaryViewSource (we have a separate class for that, which is injected/composited into the section controller and set as the supplementaryViewSource). The implementation inside that class is identical to yours, except that we want to return one size for section index 0 and another size for other indexes. Sorry for being so repetitive: in he current codebase there's no way we'll receive an index different than 0. My PR does exactly what we need, and I'm not trying to convince you in any way, I'm actually trying to understand where I'm wrong to have this bugfix working for me, fixing @MarvinNazari 's crash and not making sense for you.
This extra object parameter doesn't seem to help me, I won't use it because today I don't need it either, I need exactly what's in your code example. Try to return a different size for the second section in your example and update me.

Thanks for still going through this!